src/ConnectionAutomaton.cpp

```diff
@@ -320,13 +320,13 @@ void ConnectionAutomaton::onConnectionFrame(
   switch (type) {
     case FrameType::KEEPALIVE: {
       Frame_KEEPALIVE frame;
-      if (!deserializeFrameOrError(frame, std::move(payload))) {
+      if (!deserializeFrameOrError(isResumable_, frame, std::move(payload))) {
```

Comment:
How about this approach, store if the client cares about resumption or not (based on if resumeable if set in the SETUP frame?). I'll implement this now to show what I mean.